[docs] Add introduction sections to tutorials (#2850)
jump-dev/JuMP.jl: docs/src/tutorials/conic/introduction.md

# Introduction

[Conic programs](https://en.wikipedia.org/wiki/Conic_optimization) are a class
of convex nonlinear optimization problems which use cones to represent the
nonlinearities. They have the form:
```math
\begin{align}
    & \min_{x \in \mathbb{R}^n} & f_0(x) \\

>     & \;\;\text{s.t.} & f_i(x) \in \mathcal{S}_i & \;\; i = 1 \ldots m

Reviewer: ```suggestion
    & \;\;\text{s.t.} & f_j(x) \in \mathcal{S}_j & \;\; j = 1 \ldots m
```

Reviewer: I suggest to use `j` for constraint indices and `i` for variable indices to be consistent with the intro of LP and NLP

\end{align}
```

Mixed-integer conic programs (MICPs) are extensions of conic programs in which
some (or all) of the decision variables take discrete values.

## How to choose a solver

JuMP supports a range of conic solvers, although support differs on what types
of cones each solver supports. In the list of [Supported solvers](@ref), "SOCP"
denotes solvers supporting second-order cones and "SDP" denotes solvers

> supporting semi-definite cones. In addition, solvers such as SCS and Mosek have

Reviewer: ```suggestion
supporting semidefinite cones. In addition, solvers such as SCS and Mosek have
```

support for the exponential cone. Moreover, due to the bridging system in
MathOptInterface, many of these solvers support a much wider range of exotic
cones than they natively support. Solvers supporting discrete variables start
with "(MI)" in the list of [Supported solvers](@ref).

## How these tutorials are structured

Having a high-level overview of how this part of the documentation is structured
will help you know where to look for certain things.

 * The following tutorials are worked examples that present a problem in words,
   then formulate it in mathematics, and then solve it in JuMP. This usually
   involves some sort of visualization of the solution. Start here if you are
   new to JuMP.
   * [Experiment design](@ref)
   * [Logistic regression](@ref)
 * The [Tips and tricks](@ref conic_tips_and_tricks) tutorial contains a
   number of helpful reformulations and tricks you can use when modeling
   conic programs. Look here if you are stuck trying to formulate a problem

>    as a nonlinear program.

Reviewer: ```suggestion
   as a conic program.
```
